src/changeset/change_set.py

```diff
@@ -55,3 +72,194 @@ def __len__(self):
     def apply(self):
         for change in self.__root__:
             change.apply()
+
+
+def filter_config(defined_jobs, project_id, environment_id):
+    """Filters the config based on the inputs provided for project ids and environment ids."""
+    removed_job_ids = set()
+    if len(environment_id) != 0:
+        for job_id, job in defined_jobs.items():
+            if not job.environment_id in environment_id:
+                removed_job_ids.add(job_id)
+                logger.warning(
+                    f"For Job# {job.identifier}, environment_id(s) provided as arguments does not match the ID's in Jobs YAML file!!!"
+                )
+    if len(project_id) != 0:
+        for job_id, job in defined_jobs.items():
+            if not job.project_id in project_id:
+                removed_job_ids.add(job_id)
+                logger.warning(
+                    f"For Job# {job.identifier}, project_id(s) provided as arguments does not match the ID's in Jobs YAML file!!!"
+                )
+    return {job_id: job for job_id, job in defined_jobs.items() if job_id not in removed_job_ids}
+
+
+def build_change_set(config, disable_ssl_verification, project_id, environment_id):
+    """Compares the config of YML files versus dbt Cloud.
+    Depending on the value of no_update, it will either update the dbt Cloud config or not.
+
+    CONFIG is the path to your jobs.yml config file.
+    """
+    configuration = load_job_configuration(config)
+    unfiltered_defined_jobs = configuration.jobs
+
+    # If a project_id or environment_id is passed in as a parameter (one or multiple), check if these match the ID's in Jobs YAML file, otherwise add a warning and continue the process
+    defined_jobs = filter_config(unfiltered_defined_jobs, project_id, environment_id)
+
+    if len(defined_jobs) == 0:
+        logger.error(
+            "No jobs found in the Jobs YAML file after filtering based on the project_id and environment_id provided as arguments!!!"
+        )
```

Comment:
@b-per  - I think we can show this as a warning, as we filter out all jobs in this case, but show a success message for Plan, Sync etc;

Reply:
Modified this in the latest commit